### PR TITLE
volume profile changes

### PR DESCRIPTION
**Pull request by gnehapk**
tendrl-github-issues:
- #819 
- #788 
- #769 
- #821

### CODE REVIEW

**Review by r0h4n — APPROVED** [review]

File: src/modules/clusters/import-cluster/import-cluster.html [hunk 1]
```diff
@@ -8,12 +8,19 @@
...
             <div class="row">
-                <h3 class="col-sm-12">Import <span class="bold-text">{{vm.cluster.name}}</span> cluster to enable monitoring of the cluster.</h3>
+                <h3 class="col-sm-12">Review the hosts for the <span class="bold-text">{{vm.cluster.name}}</span> and click Import to monitor this cluster.</h3>
             </div>
             <div class="row">
                 <div class="col-sm-12 volume-profile">
-                    <input type="checkbox" name="volumeProfile" ng-model="vm.enableProfiling" />
-                    <label for="volumeProfile"> Enable Volume profiling </label><i class="pficon pficon-info" uib-tooltip-template="'../modules/clusters/import-cluster/import-cluster-tooltip.html'" tooltip-placement="bottom-left"></i>
+                    <h3 class="bold-text">Volume Profiling</h3>
+                    <input type="radio" name="enable" value="enable" id="enable" ng-model="vm.enableProfiling" checked>
```
Comment (cloudbehl) [inline src/modules/clusters/import-cluster/import-cluster.html]: If using the name property. The name property for all the input should be same as it denotes the group of a radio button.

File: src/commons/factories/volume-factory.js [hunk 2]
```diff
@@ -36,6 +36,60 @@
                 return null;
             });
         };
+
+        vm.enableProfiling = function(volume, clusterId) {
```
Comment (cloudbehl) [inline src/commons/factories/volume-factory.js]: We should make the functions generic to stop code repetition. Enable and disable profiling is having just a parameter change. so please make a generic function.

**Review by a2batic — APPROVED** [review]

File: src/commons/stores/cluster-store.js [hunk 3]
```diff
@@ -118,6 +118,25 @@
                 temp.hosts = store.getAssociatedHosts(data[i]);
                 res.push(temp);
             }
+
+            function _getProfileStatus(status) {
```
Comment (cloudbehl) [inline src/commons/stores/cluster-store.js]: Cleaner way to write:

profileState[volume_profiling_state]

var profileState = {"enable": "Enabled",
"disabled": "Disabled",
"mixed": "Mixed"}

File: src/commons/stores/volume-store.js [hunk 4]
```diff
@@ -101,6 +116,30 @@
             return null;
 
         };
+
+        store.enableProfiling = function(volume, clusterId) {
```
Comment (cloudbehl) [inline src/commons/stores/volume-store.js]: create a generic function for enabling and disable profiling.

**Review by cybernth — APPROVED** [review]

File: src/modules/tasks/global-task-detail/global-task-detail.js [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+                    startMessageTimer();
+                });
+        }
+
+        function init() {
+
+            $rootScope.selectedClusterOption = "";
+            vm.clusterId = $stateParams.clusterId;
+            if ($rootScope.clusterData) {
+                vm.clusterObj = clusterStore.getClusterDetails(vm.clusterId);
```
Comment (cloudbehl) [inline src/modules/tasks/global-task-detail/global-task-detail.js]: create a different function for the same set of lines and call a function.

basically, the below lines should in function and call the function from if and else both. will look much cleaner

+                        vm.clusterObj = clusterStore.getClusterDetails(vm.clusterId);
+                        vm.clusterName = vm.clusterObj.cluster_id || "NA";
+                        vm.clusterStatus = clusterStore.checkStatus(vm.clusterObj);
+
+                        taskStore.getJobDetail($stateParams.taskId)
+                            .then(function(data) {
+                                vm.taskDetail = data;
+                                vm.isDataLoading = false;
+
+                                _getTaskLogs();
+                                startStatusTimer();
+                                startMessageTimer();
+                            });

File: src/modules/volumes/volume-list/volume-list.html [hunk 6]
```diff
@@ -64,11 +64,18 @@ <h1>No Volumes Detected</h1>
...
+                    <div class="pull-left counts" ng-show="volume.profileStatus !== 'Pending'">{{volume.profileStatus}}</div>
+                    <div class="pull-left counts" ng-show="volume.profileStatus === 'Pending'"><a href="" ng-click="volumeCntrl.goToTaskDetail(volume)">{{volume.profileStatus}}</a></div>
+                </div>
                 <div class="ft-column list-view-pf-additional-info-item list-view-pf-additional-info-item-stacked">
                     <div class="bold-text">Alerts</div>
                     <div class="pull-left counts">{{volume.alertCount}}</div>
                 </div>
                 <div class="ft-column ft-actions">
+                    <button ng-show="volume.profileStatus === 'Enabled' || (volume.currentTask.job_name === 'StopProfiling' && volume.currentTask.status === 'in_progress')" class="btn btn-default dashboard-btn" ng-click="volumeCntrl.disableProfiling(volume, $event)" tooltip-enable="volumeCntrl.flag" ng-mouseover="volumeCntrl.addTooltip($event)" uib-tooltip="Disable Profiling" ng-disabled="volume.disableAction || volume.currentTask.status === 'in_progress'">Disable Profiling</button>
```
Comment (cloudbehl) [inline src/modules/volumes/volume-list/volume-list.html]: please return True and false to the ng-show by calling a function.

File: src/modules/volumes/volume-list/volume-list.js [hunk 7]
```diff
@@ -87,25 +90,78 @@
...
-            if (vm.sortConfig.currentField.id === "name") {
-                compValue = item1.name.localeCompare(item2.name);
-            } else if (vm.sortConfig.currentField.id === "status") {
-                compValue = item1.status.localeCompare(item2.status);
-            }
+        function disableProfiling(volume, $event) {
 
-            if (!vm.sortConfig.isAscending) {
-                compValue = compValue * -1;
+            volumeStore.disableProfiling(volume, vm.clusterId)
```
Comment (cloudbehl) [inline src/modules/volumes/volume-list/volume-list.js]: same as above,

create a single and generic function for both enable and disable

File: src/modules/volumes/volume-list/volume-list.js [hunk 8]
```diff
@@ -164,48 +220,25 @@
...
-
-        /* Trigger this function when we have cluster data */
-        $scope.$on("GotClusterData", function(event, data) {
-            /* Forward to home view if we don't have any cluster */
-            if ($rootScope.clusterData === null || $rootScope.clusterData.clusters.length === 0) {
-                $state.go("clusters");
-            } else {
-                init();
+        function _compareFn(item1, item2) {
+            var compValue = 0;
```
Comment (cloudbehl) [inline src/modules/volumes/volume-list/volume-list.js]: can avoid If else by doing the below.

var sortId = vm.sortConfig.currentField.id;
compValue = item1["sortId"].localeCompare(item2[sortId]);

**Review by cloudbehl — CHANGES_REQUESTED** [review]
Please make the required changes.

File: src/modules/volumes/volume-list/volume-list.js (hunk 8 above)
Comment (a2batic) [reply src/modules/volumes/volume-list/volume-list.js]: @cloudbehl, but wont this function loose its consistency when compared to other views ?

Comment (cloudbehl) [reply src/modules/volumes/volume-list/volume-list.js]: @a2batic Take it as a refactoring for other views. If it improves this we can work with other views and maintain that consistency

**Review by cloudbehl — APPROVED** [review]
LGTM

**Pull request merged**